### ISSUE TITLE
getKet duplicates items when macro molucules are used

### ISSUE THREAD

**Issue by doktordirk** [epam/ketcher#5181]
**Steps to Reproduce**
1. Add item in micro mode
2. Switch to macro mode
3. Add item in macro mode
4. Switch to micro mode
5. Get result from `ketcher.getKet()`

**Actual behavior**
All items in the ket data are duplicated

**Expected behavior**
No item duplicated. Output should be identical to using "Save as..." -> ket file

**Screenshots**

<details>
<summary>output of "Save as..." -> ket file</summary>

```
{
    "root": {
        "nodes": [
            {
                "$ref": "monomer19"
            },
            {
                "$ref": "mol0"
            }
        ],
        "connections": [],
        "templates": [
            {
                "$ref": "monomerTemplate-Cya___3-sulfoalanine"
            }
        ]
    },
    "mol0": {
        "type": "molecule",
        "atoms": [
            {
                "label": "C",
                "location": [
                    10.208987298340658,
                    -8.74999266647844,
                    0
                ]
            },
            {
                "label": "C",
                "location": [
                    10.208987298340658,
                    -9.750007333521562,
                    0
                ]
            },
            {
                "label": "C",
                "location": [
                    11.075000000000001,
                    -10.250014667043123,
                    0
                ]
            },
            {
                "label": "C",
                "location": [
                    11.941012701659346,
                    -9.750007333521562,
                    0
                ]
            },
            {
                "label": "C",
                "location": [
                    11.941012701659346,
                    -8.74999266647844,
                    0
                ]
            },
            {
                "label": "C",
                "location": [
                    11.075000000000001,
                    -8.249985332956879,
                    0
                ]
            }
        ],
        "bonds": [
            {
                "type": 1,
                "atoms": [
                    5,
                    0
                ]
            },
            {
                "type": 1,
                "atoms": [
                    0,
                    1
                ]
            },
            {
                "type": 1,
                "atoms": [
                    1,
                    2
                ]
            },
            {
                "type": 1,
                "atoms": [
                    2,
                    3
                ]
            },
            {
                "type": 1,
                "atoms": [
                    3,
                    4
                ]
            },
            {
                "type": 1,
                "atoms": [
                    4,
                    5
                ]
            }
        ]
    },
    "monomer19": {
        "type": "monomer",
        "id": "19",
        "position": {
            "x": 13.5,
            "y": -6.775
        },
        "alias": "Cya",
        "templateId": "Cya___3-sulfoalanine"
    },
    "monomerTemplate-Cya___3-sulfoalanine": {
        "type": "monomerTemplate",
        "atoms": [
            {
                "label": "C",
                "location": [
                    0.5437,
                    0.778,
                    0
                ]
            },
            {
                "label": "C",
                "location": [
                    0.5429,
                    -0.7228,
                    0
                ],
                "stereoLabel": "abs"
            },
            {
                "label": "C",
                "location": [
                    1.8417,
                    -1.4748,
                    0
                ]
            },
            {
                "label": "O",
                "location": [
                    1.841,
                    -2.6748,
                    0
                ]
            },
            {
                "label": "O",
                "location": [
                    2.8815,
                    -0.8757,
                    0
                ]
            },
            {
                "label": "N",
                "location": [
                    -0.756,
                    -1.4748,
                    0
                ]
            },
            {
                "label": "H",
                "location": [
                    -1.7964,
                    -0.8767,
                    0
                ]
            },
            {
                "label": "S",
                "location": [
                    -0.7551,
                    1.53,
                    0
                ]
            },
            {
                "label": "O",
                "location": [
                    -0.7544,
                    2.73,
                    0
                ]
            },
            {
                "label": "O",
                "location": [
                    -1.7949,
                    0.9309,
                    0
                ]
            },
            {
                "label": "O",
                "location": [
                    -1.794,
                    2.1307,
                    0
                ]
            }
        ],
        "bonds": [
            {
                "type": 1,
                "atoms": [
                    1,
                    0
                ],
                "stereo": 1
            },
            {
                "type": 1,
                "atoms": [
                    1,
                    2
                ]
            },
            {
                "type": 2,
                "atoms": [
                    2,
                    3
                ]
            },
            {
                "type": 1,
                "atoms": [
                    2,
                    4
                ]
            },
            {
                "type": 1,
                "atoms": [
                    1,
                    5
                ]
            },
            {
                "type": 1,
                "atoms": [
                    5,
                    6
                ]
            },
            {
                "type": 1,
                "atoms": [
                    0,
                    7
                ]
            },
            {
                "type": 2,
                "atoms": [
                    7,
                    8
                ]
            },
            {
                "type": 2,
                "atoms": [
                    7,
                    9
                ]
            },
            {
                "type": 1,
                "atoms": [
                    7,
                    10
                ]
            }
        ],
        "class": "AminoAcid",
        "classHELM": "PEPTIDE",
        "id": "Cya___3-sulfoalanine",
        "fullName": "3-sulfoalanine",
        "alias": "Cya",
        "attachmentPoints": [
            {
                "attachmentAtom": 5,
                "leavingGroup": {
                    "atoms": [
                        6
                    ]
                },
                "type": "left"
            },
            {
                "attachmentAtom": 2,
                "leavingGroup": {
                    "atoms": [
                        4
                    ]
                },
                "type": "right"
            }
        ],
        "naturalAnalogShort": "A"
    }
}
```

</details>

<details>
<summary>output of ketcher.getKet()</summary>

{
    "root": {
        "nodes": [
            {
                "$ref": "monomer11"
            },
            {
                "$ref": "monomer21"
            },
            {
                "$ref": "mol0"
            },
            {
                "$ref": "mol1"
            }
        ],
        "connections": [],
        "templates": [
            {
                "$ref": "monomerTemplate-Cya___3-sulfoalanine"
            }
        ]
    },
    "mol0": {
        "type": "molecule",
        "atoms": [
            {
                "label": "C",
                "location": [
                    10.208987298340658,
                    -8.74999266647844,
                    0
                ]
            },
            {
                "label": "C",
                "location": [
                    10.208987298340658,
                    -9.750007333521562,
                    0
                ]
            },
            {
                "label": "C",
                "location": [
                    11.075000000000001,
                    -10.250014667043123,
                    0
                ]
            },
            {
                "label": "C",
                "location": [
                    11.941012701659346,
                    -9.750007333521562,
                    0
                ]
            },
            {
                "label": "C",
                "location": [
                    11.941012701659346,
                    -8.74999266647844,
                    0
                ]
            },
            {
                "label": "C",
                "location": [
                    11.075000000000001,
                    -8.249985332956879,
                    0
                ]
            }
        ],
        "bonds": [
            {
                "type": 1,
                "atoms": [
                    5,
                    0
                ]
            },
            {
                "type": 1,
                "atoms": [
                    0,
                    1
                ]
            },
            {
                "type": 1,
                "atoms": [
                    1,
                    2
                ]
            },
            {
                "type": 1,
                "atoms": [
                    2,
                    3
                ]
            },
            {
                "type": 1,
                "atoms": [
                    3,
                    4
                ]
            },
            {
                "type": 1,
                "atoms": [
                    4,
                    5
                ]
            }
        ]
    },
    "mol1": {
        "type": "molecule",
        "atoms": [
            {
                "label": "C",
                "location": [
                    10.208987298340658,
                    -8.74999266647844,
                    0
                ]
            },
            {
                "label": "C",
                "location": [
                    10.208987298340658,
                    -9.750007333521562,
                    0
                ]
            },
            {
                "label": "C",
                "location": [
                    11.075000000000001,
                    -10.250014667043123,
                    0
                ]
            },
            {
                "label": "C",
                "location": [
                    11.941012701659346,
                    -9.750007333521562,
                    0
                ]
            },
            {
                "label": "C",
                "location": [
                    11.941012701659346,
                    -8.74999266647844,
                    0
                ]
            },
            {
                "label": "C",
                "location": [
                    11.075000000000001,
                    -8.249985332956879,
                    0
                ]
            }
        ],
        "bonds": [
            {
                "type": 1,
                "atoms": [
                    5,
                    0
                ]
            },
            {
                "type": 1,
                "atoms": [
                    0,
                    1
                ]
            },
            {
                "type": 1,
                "atoms": [
                    1,
                    2
                ]
            },
            {
                "type": 1,
                "atoms": [
                    2,
                    3
                ]
            },
            {
                "type": 1,
                "atoms": [
                    3,
                    4
                ]
            },
            {
                "type": 1,
                "atoms": [
                    4,
                    5
                ]
            }
        ]
    },
    "monomer11": {
        "type": "monomer",
        "id": "11",
        "position": {
            "x": 13.5,
            "y": -6.775
        },
        "alias": "Cya",
        "templateId": "Cya___3-sulfoalanine"
    },
    "monomerTemplate-Cya___3-sulfoalanine": {
        "type": "monomerTemplate",
        "atoms": [
            {
                "label": "C",
                "location": [
                    0.5437,
                    0.778,
                    0
                ]
            },
            {
                "label": "C",
                "location": [
                    0.5429,
                    -0.7228,
                    0
                ],
                "stereoLabel": "abs"
            },
            {
                "label": "C",
                "location": [
                    1.8417,
                    -1.4748,
                    0
                ]
            },
            {
                "label": "O",
                "location": [
                    1.841,
                    -2.6748,
                    0
                ]
            },
            {
                "label": "O",
                "location": [
                    2.8815,
                    -0.8757,
                    0
                ]
            },
            {
                "label": "N",
                "location": [
                    -0.756,
                    -1.4748,
                    0
                ]
            },
            {
                "label": "H",
                "location": [
                    -1.7964,
                    -0.8767,
                    0
                ]
            },
            {
                "label": "S",
                "location": [
                    -0.7551,
                    1.53,
                    0
                ]
            },
            {
                "label": "O",
                "location": [
                    -0.7544,
                    2.73,
                    0
                ]
            },
            {
                "label": "O",
                "location": [
                    -1.7949,
                    0.9309,
                    0
                ]
            },
            {
                "label": "O",
                "location": [
                    -1.794,
                    2.1307,
                    0
                ]
            }
        ],
        "bonds": [
            {
                "type": 1,
                "atoms": [
                    1,
                    0
                ],
                "stereo": 1
            },
            {
                "type": 1,
                "atoms": [
                    1,
                    2
                ]
            },
            {
                "type": 2,
                "atoms": [
                    2,
                    3
                ]
            },
            {
                "type": 1,
                "atoms": [
                    2,
                    4
                ]
            },
            {
                "type": 1,
                "atoms": [
                    1,
                    5
                ]
            },
            {
                "type": 1,
                "atoms": [
                    5,
                    6
                ]
            },
            {
                "type": 1,
                "atoms": [
                    0,
                    7
                ]
            },
            {
                "type": 2,
                "atoms": [
                    7,
                    8
                ]
            },
            {
                "type": 2,
                "atoms": [
                    7,
                    9
                ]
            },
            {
                "type": 1,
                "atoms": [
                    7,
                    10
                ]
            }
        ],
        "class": "AminoAcid",
        "classHELM": "PEPTIDE",
        "id": "Cya___3-sulfoalanine",
        "fullName": "3-sulfoalanine",
        "alias": "Cya",
        "attachmentPoints": [
            {
                "attachmentAtom": 5,
                "leavingGroup": {
                    "atoms": [
                        6
                    ]
                },
                "type": "left"
            },
            {
                "attachmentAtom": 2,
                "leavingGroup": {
                    "atoms": [
                        4
                    ]
                },
                "type": "right"
            }
        ],
        "naturalAnalogShort": "A"
    },
    "monomer21": {
        "type": "monomer",
        "id": "21",
        "position": {
            "x": 13.5,
            "y": -6.775
        },
        "alias": "Cya",
        "templateId": "Cya___3-sulfoalanine"
    }
}

</details>

**Desktop (please complete the following information):**
 - OS: Ubuntu 
 - Browser: chrome
 - Version : 127.0.6533.72 (Official Build) (64-bit)

**Ketcher version** 
2.22.0-rc.9

**Additional context**
using
```
const structOrString = await window.ketcher.editor.struct();
const { struct } = await window.ketcher.indigo.convert(structOrString, {
        outputFormat: chemical/x-indigo-ket',
      });
```
gives the correct ket data (but has other issues since converting (basically) ket to ket flattes arrows

**Comment by rrodionov91** [comment]
Hi @doktordirk 
Thank you for highlighting the issue. We reproduced it. Will plan the fix.

**Comment by NataliaLoginova** [comment]
This issue was solved in PR [#5181 - getKet duplicates items when macro molucules are use](https://github.com/epam/ketcher/pull/5509)

**Comment by Zhirnoff** [comment]
Tested. Bug fixed.
![2024-11-08_10h47_28](https://github.com/user-attachments/assets/309b7d20-6992-4e3d-954f-96c4126dc8c4)


**Desktop:**
 - OS: Windows 11
 - Browser Chrome
 - Version 128.0.6613.120 (Official Build) (64-bit)

**Ketcher version** 
[Version 2.27.0-rc.1]
**Indigo version** 
[Version 1.26.0-rc.1]
